ClientApp/viking-vault/src/AttachCardModal.tsx

```diff
@@ -0,0 +1,215 @@
+import React from "react";
+import { constants } from "./ConstantVariables";
+import './AttachCardModal.css';
+
+const baseUrl = constants.baseUrl;
+let regexCheckIfOnlyDigits = /^([+-]?[1-9]\d*|0)$/;
+let currentYearValue = new Date().getFullYear();
+
+interface IModalProps {
+    open: boolean;
+    onModalClose: any;
+    firstName: string;
+    lastName: string;
+    userId: number;
+}
+
+interface ICardProperties {
+    cardNumber: string;
+    expirationDate: string;
+    ccv: string;
+}
+
+interface IFormState {
+    [key: string]: string | boolean | ICardProperties  | null;
+    card: ICardProperties;
+    valid: boolean | null;
+    errorLabel: string;
+}
+
+class AttachCardForm extends React.Component<any, IFormState> {
+    constructor(props: IModalProps) {
+        super(props);
+        this.state = {
+            card: {
+                cardNumber: "",
+                expirationDate: "",
+                ccv: "",
+            },
+            valid: null,
+            errorLabel: "",
+        }
+    }
+
+    private handleChange = (inputValue: string, inputName: string) => {
+        this.setState({
+            card: {
+                ...this.state.card,
+                [inputName]: inputValue
+            }
+        });
+    }
+
+    private cardNumberHasSixteenDigits = () => {
+        if(this.state.card.cardNumber.length === 16 && this.state.card.cardNumber.match(regexCheckIfOnlyDigits))
+            return  {
+                message: "Ok",
+                class: "alert alert-success"
+            }
+        return  {
+            message: "Card Number must have 16 digits",
+            class: "alert alert-danger"
+        }
+    }
+
+    private ccvHasThreeDigits = () => {
+        if(this.state.card.ccv.match(regexCheckIfOnlyDigits) && this.state.card.ccv.length === 3)
+            return  {
+                message: "Ok",
+                class: "alert alert-success"
+            }
+        return  {
+            message: "CCV must have 3 digits",
+            class: "alert alert-danger"
+        }
+    }
+
+    private expirationDateIsAfterCurrentYear = () => {
+        if(!(this.state.card.expirationDate === ""))
+            return  {
+                message: "Ok",
+                class: "alert alert-success"
+            }
+        return  {
+            message: "Expiration Date must be after at least one year from the current date!",
+            class: "alert alert-danger"
+        }
+    }
+
+    private mandatoryFieldsCompletedCorrectly = () => {
+        let val = true;
+        if(this.cardNumberHasSixteenDigits().message !== "Ok") {
```

Comment:
I'd replace all of this to something like this:


const validationFunctions = [this.cardNumberHasSixteenDigits, this.expirationDateIsAfterCurrentYear, this.ccvHasThreeDigits];

let val = false;

val = validationFunctions.all(functionItem => functionItem().message === "Ok")

return val;

In this way if you wanna add another validation function, you simply add it to the array (this actual code logic might be broken, but the idea should be that you store the validation functions in an array and then loop over them, call them and check for the `.message`